include/kernel.h

```diff
@@ -4191,6 +4191,17 @@ extern void k_free(void *ptr);
  */
 extern void *k_calloc(size_t nmemb, size_t size);
 
+/**
+ * @brief Allocate memory from system heap.
+ *
+ * Such memory must be released using k_free().
+ *
+ * @param size Amount of memory requested (in bytes).
+ *
+ * @return Address of the allocated memory if successful; otherwise NULL.
+ * @req K-HEAP-001
+ */
+extern void *z_thread_malloc(size_t size);
```

Comment:
Why are you doing this????

This is an internal kernel API and it is already declared in kernel/include/kernel_internal.h!

Reply:
@andrewboie Sorry, I did not check declaration in this file.
Included <kernel_internal.h>, in the test case to resolve implicit declaration of function error.